pkg/image/api/v1/types.go

```diff
@@ -144,3 +182,44 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+type ImageStreamImport struct {
+	unversioned.TypeMeta `json:",inline"`
+	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
+
+	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
+	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
+}
+
+type RepositoryImport struct {
+	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
+	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
```

Comment:
That's intentional.

On Mon, Jan 11, 2016 at 1:21 PM, David Eads <notifications@github.com>
wrote:

> In pkg/image/api/v1/types.go
> <https://github.com/openshift/origin/pull/6288#discussion_r49356863>:
>
> > @@ -144,3 +182,44 @@ type DockerImageReference struct {
> >  	Tag       string
> >  	ID        string
> >  }
> > +
> > +type ImageStreamImport struct {
> > +	unversioned.TypeMeta `json:",inline"`
> > +	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
> > +
> > +	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
> > +	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
> > +}
> > +
> > +type RepositoryImport struct {
> > +	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
> > +	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
>
> If I'm reading correctly, this means that a default ImageStreamImport
> object doesn't actually do an import? That seems backwards.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6288/files#r49356863>.
>